apis/Google.Cloud.EntityFrameworkCore.Spanner/Google.Cloud.EntityFrameworkCore.Spanner/Update/Internal/SpannerModificationCommandBatch.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Api.Gax;
+using Google.Cloud.Spanner.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.EntityFrameworkCore.Update;
+
+namespace Google.Cloud.EntityFrameworkCore.Spanner.Update.Internal
+{
+    /// <summary>
+    /// </summary>
+    public class SpannerModificationCommandBatch : ModificationCommandBatch
+    {
+        private readonly List<ModificationCommand> _modificationCommands = new List<ModificationCommand>();
+        private readonly IRelationalTypeMapper _typeMapper;
+
+        /// <summary>
+        /// </summary>
+        /// <param name="typeMapper"></param>
+        public SpannerModificationCommandBatch(IRelationalTypeMapper typeMapper) => _typeMapper = typeMapper;
+
+        /// <summary>
+        /// </summary>
+        public override IReadOnlyList<ModificationCommand> ModificationCommands => _modificationCommands;
+
+        /// <summary>
+        /// </summary>
+        /// <param name="modificationCommand"></param>
+        /// <returns></returns>
+        public override bool AddCommand(ModificationCommand modificationCommand)
+        {
+            _modificationCommands.Add(modificationCommand);
+            return true;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="connection"></param>
+        public override void Execute(IRelationalConnection connection)
+        {
+            //TODO(benwu): call sync down the stack.
+            ExecuteAsync(connection).WaitWithUnwrappedExceptions();
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public override async Task ExecuteAsync(IRelationalConnection connection,
+            CancellationToken cancellationToken = new CancellationToken())
```

Comment:
I'm just matching the base method (I get a linter warning about it).  It might be equivalent after the compiler has a run at it, but I don't feel the need to tempt fate.